From d6a60496089fcd72854eb791412052cc34b74816 Mon Sep 17 00:00:00 2001
From: "Thakur, Santosh Krx" <santosh.krx.thakur@intel.com>
Date: Thu, 11 Feb 2016 20:47:53 +0530
Subject: [PATCH 1/6] SoC: Intel: BYT: klockwork error in Audio Driver

This patch checks for NULL before dereferencing "mc_link".

Change-Id: I91d39ac8fa003e96ef61b99c6a804980fd042713
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-19824
Signed-off-by: Thakur, Santosh Krx <santosh.krx.thakur@intel.com>
Reviewed-on: https://android.intel.com:443/469325
---
 sound/soc/intel/board/byt_cr_board_configs.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sound/soc/intel/board/byt_cr_board_configs.c b/sound/soc/intel/board/byt_cr_board_configs.c
index 94aa7b7..7a892e0 100644
--- a/sound/soc/intel/board/byt_cr_board_configs.c
+++ b/sound/soc/intel/board/byt_cr_board_configs.c
@@ -80,6 +80,8 @@ const struct board_config *get_board_config(
 {
 	const struct dmi_system_id *dmi;
 
+	if (!mc_link)
+		return NULL;
 	dmi = dmi_first_match(mc_link->dmi_system_ids);
 	if (!dmi) {
 		pr_info("Board is a default %s.\n", mc_link->mach_dev->name);
-- 
1.9.1

